app/controllers/close_contacts_controller.rb

```diff
@@ -2,12 +2,37 @@
 
 # CloseContactsController: close contacts
 class CloseContactsController < ApplicationController
+  include CloseContactQueryHelper
+
   before_action :authenticate_user!
   before_action :check_can_create, only: %i[create]
   before_action :check_can_edit, only: %i[update destroy]
   before_action :check_patient
   before_action :check_close_contact, only: %i[update destroy]
 
+  def index
+    redirect_to(root_url) && return unless current_user&.can_view_patient_close_contacts?
+
+    # Validate params and handle errors if invalid
+    begin
+      data = validate_close_contact_query(params)
+    rescue StandardError => e
+      render(json: { error: e.message }, status: :bad_request) && return
+    end
+
+    # Verify user has access to patient, patient exists, and the patient has close_contacts
+    patient = current_user.get_patient(data[:patient_id])
```

Comment:
Sorry for making you change this part of the code again, but I think we can actually simplify this by adding `index` to the `:check_patient` hook and referring to `patient` here as `@patient` instead. Can you also make this change for the `VaccinesController` and any of the other ones that have similar logic?

Reply:
Good idea! Making the change in both